### PR TITLE
OpenAPI update, PUT and HEAD methods in /api/v1/files/{storage}/{path} endpoint

### PR DESCRIPTION
**Pull request by Majkluss**

### CODE REVIEW

File: spec/openapi.yaml [hunk 1]
```diff
@@ -83,10 +96,43 @@ paths:
...
           $ref: "#/components/responses/NotFound"
+        409:
+          $ref: "#/components/responses/Conflict"
+    head:
+      summary: file presence and state check
+      responses:
+        200:
+          description: OK
+          content:
+            application/octet-stream:
```
Comment (ondratu) [inline spec/openapi.yaml:123]: HEAD does not return content - ever!

File: spec/openapi.yaml [hunk 2]
```diff
@@ -83,10 +96,43 @@ paths:
...
+              schema:
+                type: object
+                properties:
+                  ro:
```
Comment (ondratu) [inline spec/openapi.yaml:127]: This must be in Header value....

File: spec/openapi.yaml [hunk 3]
```diff
@@ -157,6 +203,31 @@ components:
         - path
         - type
 
+    UploadFile:
```
Comment (ondratu) [inline spec/openapi.yaml:206]: Not needed

**Review by ondratu — CHANGES_REQUESTED** [review]

File: spec/openapi.yaml [hunk 4]
```diff
@@ -58,22 +58,32 @@ paths:
...
+          example: "/local"
       - in: path
         name: path
-        required: true
         description: Path to the file
+        required: true
         schema:
           type: string
-          example: "PrusaLink gcodes/Spice_Harvester_0.3mm_PLA_MK3S_12m.gcode"
+          example: "/local/Spice_Harvester_0.3mm_PLA_MK3S_12m.gcode"
```
Comment (vorner) [inline spec/openapi.yaml:86]: This one is probably bit technical, but if we say that the example of `path` is `/local/something`, then the whole URL would be `/local//local/...`. I think the original example was correct.

File: spec/openapi.yaml [hunk 5]
```diff
@@ -83,10 +93,60 @@ paths:
...
+          description: Length of file to upload
+          schema:
+            type: integer
+            example: 101342
+        - in: header
+          name: Content-Type
+          description: Type of uploaded media
+          schema:
+            type: string
+            default: application/binary
```
Comment (vorner) [inline spec/openapi.yaml:110]: Also technical nitpick. Is there such a mime type? Aren't you confusing it with application/octet-stream?

Comment (ondratu) [reply spec/openapi.yaml:110]: `default` means - when client did not set. Or we can require this header, but with different response: 400 - Bad Request.
* bad mime type - 415 Unsupported Media Type
* no required header - 400 Bad Request

**Review by ondratu — APPROVED** [review]

**Pull request merged**